Edits to JS Getting Started (#2160)
getsentry/sentry-docs: src/platforms/javascript/index.mdx

  - browser
---

> On this page, we get you up and running with Sentry's JavaScript SDK, automatically reporting errors and exceptions in your application.

Reviewer: ```suggestion
On this page, we get you up and running with Sentry's JavaScript SDK, so that it will automatically report errors and exceptions in your application.
```

$ npm install --save @sentry/browser
```

<Alert>
<markdown>

If you are using CDN to install our SDK, please see our documentation for [installing with CDN](/platforms/javascript/configuration/install-cdn/).

> If you have Content Security Policy (CSP) set up on your site, please see [our documentation about adding to the `script-src`](/platforms/javascript/troubleshooting/).

Reviewer: I believe this only applies if you're using the CDN, so now that the CDN stuff is on its own page, I think this can disappear from here.

Author: agreed. I thought I had removed it. Thanks for the catch.

Author: Changed this to "If you are using CDN to install our SDK, please see our documentation for [installing with CDN](/platforms/javascript/configuration/install-cdn/)"

Reviewer: But it's still here...

Author: Just pushed the final changes :)

  crossorigin="anonymous"
></script>
```
<Alert>
<markdown>

Using `defer`

If you include `defer` in your script tag, any errors that occur in scripts executed before the browser SDK script executes won’t be caught, since the SDK won’t yet be initialized. We strongly recommend that if you use `defer`, you both place the script tag for the browser SDK first and also mark it -- and all of your other scripts -- `defer` (but not `async`), guaranteeing that it’s executed before any of the others.

</markdown>
</Alert>

**Note:** If you have Content Security Policy (CSP) set up on your site, please see [our documentation about adding to the `script-src`](/platforms/javascript/common/troubleshooting/).

## Configure and Verify

> `init` Sentry as soon as possible and include the verification of your setup with an intentionally sent error that breaks your application. For example, in JavaScript, calling an undefined function throws an exception::

Reviewer: ```suggestion
`init` Sentry as soon as possible in your app, to make sure it can monitor everything that follows it. Once this is done, all unhandled exceptions are automatically captured by Sentry.

The snippet below includes an intentional error, so you can test that everything is working as soon as you set it up. To see the recorded error, log into sentry.io and open your project. Clicking on the error's title will open a page where you can see detailed information and mark it as resolved.
```

```javascript {tabTitle: ESM}
import * as Sentry from "@sentry/browser";

Sentry.init({ dsn: "___PUBLIC_DSN___" });

> myUndefinedFunction(); // To test your setup

Reviewer: ```suggestion

myUndefinedFunction(); // To test your setup
```

```

```javascript {tabTitle: CDN}
Sentry.init({ dsn: "___PUBLIC_DSN___" });

> myUndefinedFunction(); // To test your setup

Reviewer: ```suggestion

myUndefinedFunction(); // To test your setup
```

```

> Once this is done, all unhandled exceptions are automatically captured by Sentry.

Reviewer: ```suggestion
```

Moved above snippet.

> **Important:** Note your DSN. The _DSN_ (Data Source Name) tells the SDK where to send events. If you forget it, view _Settings -> Projects -> Client Keys (DSN)_ in the Sentry web UI.

Reviewer: ```suggestion
**Important:** Note your DSN. The _DSN_ (Data Source Name) tells the SDK where to send events. If you forget it, go to _Settings -> Projects -> Your Project -> Client Keys (DSN)_ in the Sentry web UI.
```

> Resolve the created error by logging in to sentry.io, then opening your project. On the **Issue Details** page, scroll down to the Exception stack trace and view the error.

Reviewer: ```suggestion
```

Moved above snippet.

## Capture Errors

<Alert>
<markdown>
Key terms:

- An _event_ is one instance of sending data to Sentry. Generally, this data is an error or exception.
- An _issue_ is a grouping of similar events.
- The reporting of an event is called _capturing_.
  When an event is captured, it’s sent to Sentry.

> </Alert>
> </markdown>

Reviewer 2: ```suggestion
</markdown>
</Alert>
```

By including and configuring Sentry, our JavaScript SDK automatically attaches global handlers to _capture_ uncaught exceptions and unhandled promise rejections, as described in the official ECMAScript 6 standard. You can disable this default behavior by changing the `onunhandledrejection` option to `false` in your GlobalHandlers integration and manually hook into each event handler, then call `Sentry.captureException` or `Sentry.captureMessage` directly.

> You may also need to manage your configuration if you are using a third-party library to implement promises. Also, remember browsers often implement security measures that can block error reporting when serving script files from different origins.

Reviewer: ```suggestion
You may also need to manage your configuration if you are using a third-party library to implement promises. Also, remember that browsers often implement security measures that can block error reporting when serving script files from different origins.
```

});
```

Learn more about what releases can do in [our documentation for releases](/product/releases/).

### Upload Source Maps

We **highly recommend** you incorporate source maps to receive the full benefit of error tracking and monitoring. See our [source maps documentation](/platforms/javascript/configuration/sourcemaps) to learn more.

## Monitor Performance

> Install performance monitoring using Sentry’s JavaScript SDK using both the `@sentry/browser` and `@sentry/tracing` packages, or link directly to our combined minified bundle via CDN:

Reviewer: ```suggestion
Install performance monitoring using Sentry’s JavaScript SDK using both the `@sentry/browser` and `@sentry/tracing` packages:
```

$ npm install --save @sentry/browser @sentry/tracing
```

<Alert level="note" title="Note"><markdown>

We switched our package name from `@sentry/apm` to `@sentry/tracing`. If you were previously using `@sentry/apm`, please see [Switching to @sentry/tracing](/product/performance/apm-to-tracing/).

> If you are using CDN, please see our documentation for [installing with CDN](/platforms/javascript/configuration/install-cdn).

Reviewer: ```suggestion
If you are using our CDN, please see our documentation for [installing with CDN](/platforms/javascript/configuration/install-cdn).
```

$ npm install --save @sentry/browser @sentry/tracing
```

> <Alert level="Note" title="Note"><markdown>

Reviewer 2: ```suggestion
<Alert level="note" title="Note"><markdown>
```
